### DISCUSSION TITLE
Shadows disappear when looking from afar

### DISCUSSION THREAD

**Discussion by yavuz650** [google/filament#6486] [Q&A]
I'm rendering the PBR Sponza scene and I added some point lights to it using the following configuration,
`

      filament::LightManager::Builder(filament::LightManager::Type::POINT)
            .color(lm.getColor(lm.getInstance(entities[i])))
            .intensity(50000.0f, filament::LightManager::EFFICIENCY_LED)
            .position(pos)
            .falloff(10)
            .castShadows(true)
            .build(*engine, entities[i]);
`

Now, the problem is, the lights and the shadows look as expected when the camera is near them. When I move away, the shadows disappear, as shown in the images below,

![image](https://user-images.githubusercontent.com/24359352/214403675-0c4dc86a-179d-4341-a566-7af3590b003f.png)

![image](https://user-images.githubusercontent.com/24359352/214403590-e64729d4-092d-485f-ba96-b468e3df9cb0.png)

I played with `ShadowOptions` (`shadowFar`), but that didn't help. How can I render shadows at every distance?

**Comment by romainguy** [comment]
You can control this using View::setDynamicLightingOptions and setting an appropriate "light far". Note that this will affect performance.

**Reply by yavuz650** [reply]
I just tried this, but that didn't help. From what I understand, this setting turns on/off the lights based on the given near/far. It did not affect the shadows, however.
Just to make sure, I set `lightFar` to a very high value,
`view->setDynamicLightingOptions(1.0, 300000.0);`. Didn't work.
